examples/nlp/language_modeling/megatron_ckpt_to_nemo.py

```diff
@@ -82,91 +87,100 @@ def get_args():
-    )
-
-    app_state.pipeline_model_parallel_rank = parallel_state.get_pipeline_model_parallel_rank()
-    app_state.tensor_model_parallel_rank = parallel_state.get_tensor_model_parallel_rank()
+    save_connector = NLPSaveRestoreConnector()
+    trainer = Trainer(devices=1, strategy=NLPDDPStrategy(), accelerator="cpu")
 
-    # inject model parallel rank
     checkpoint_path = inject_model_parallel_rank(os.path.join(args.checkpoint_folder, args.checkpoint_name))
-
-    logging.info(
-        f'rank: {rank}, local_rank: {local_rank}, is loading checkpoint: {checkpoint_path} for tp_rank: {app_state.tensor_model_parallel_rank} and pp_rank: {app_state.pipeline_model_parallel_rank}'
-    )
-
     if args.model_type == 'gpt':
-        model = MegatronGPTModel.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+        model = MegatronGPTModel.load_from_checkpoint(
+            checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
+        ).to("cpu")
     elif args.model_type == 'bert':
         model = MegatronBertModel.load_from_checkpoint(
             checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
-        )
+        ).to("cpu")
     elif args.model_type == 't5':
-        model = MegatronT5Model.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+        model = MegatronT5Model.load_from_checkpoint(
+            checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
+        ).to("cpu")
     elif args.model_type == 'bart':
         model = MegatronBARTModel.load_from_checkpoint(
             checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
-        )
+        ).to("cpu")
     elif args.model_type == 'nmt':
-        model = MegatronNMTModel.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+        model = MegatronNMTModel.load_from_checkpoint(
+            checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
+        ).to("cpu")
     elif args.model_type == 'retro':
         model = MegatronRetrievalModel.load_from_checkpoint(
             checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
-        )
-    model._save_restore_connector = NLPSaveRestoreConnector()
+        ).to("cpu")
+
+    checkpoint_dir = args.checkpoint_folder
+    checkpoint_name = args.checkpoint_name
+    save_path = os.path.dirname(args.nemo_file_path)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+
+        if app_state.pipeline_model_parallel_size == 1:
+            # move weights to the tmpdir
+            for tp_rank in range(app_state.tensor_model_parallel_size):
+                os.makedirs(os.path.join(tmpdir, f'mp_rank_{tp_rank:02d}'))
+                mp_model_weights = os.path.join(checkpoint_dir, f'mp_rank_{tp_rank:02d}', checkpoint_name)
+                # TODO: Change to a move
+                shutil.copy(
```

Comment:
Copy then rm?

Reply:
Copy then RM makes more sense here.  If on the above comment I can just call save_to then I think this might be irrelevant anyway.

Reply:
I think this is ready for someone to take a look at. I just realized that because I have the copy under a tmp directory its actually like a copy+rm operation already.